server/methods/core/vendor.js

```diff
@@ -29,17 +29,15 @@ Meteor.methods({
     });
   },
   "vendor/getVendorId"() {
-    const profile = Collections.Accounts.find({userId: Meteor.userId()}).fetch();
-    if (profile.length > 0 && profile[0].profile !== undefined && profile[0].profile.vendorDetails !== undefined) {
-      return (profile[0].profile.vendorDetails[0]._id);
+    const Profile = Collections.Accounts.findOne({ userId: Meteor.userId() }).fetch();
```

Comment:
The naming convention is confusing. I used `Profile` as an example. Please choose a better name declaration.